support-services/src/main/scala/com/gu/support/promotions/PromotionApplicator.scala

```diff
@@ -55,6 +58,56 @@ class DiscountApplicator(discount: DiscountBenefit, config: PromotionsDiscountCo
   )
 }
 
+class IntroductoryPriceApplicator(introductoryPriceBenefit: IntroductoryPriceBenefit) extends BenefitApplicator {
+  def applyTo(subscriptionData: SubscriptionData): SubscriptionData = {
+    val subscription = subscriptionData.subscription
+    val result = introductoryPriceRatePlanData(subscriptionData).map(
+      ratePlanData =>
+        subscriptionData.copy(
+          ratePlanData = subscriptionData.ratePlanData.::(ratePlanData),
+          subscription = subscription.copy(
+            contractAcceptanceDate = subscription.contractAcceptanceDate.plusWeeks(introductoryPriceBenefit.periodLength)
```

Comment:
so if `introductoryPrice` exists on the promo code,we try to apply the applicator.  We add the period length to the start date (which would hopefully always be 6 at the moment, and we override the price in the charge from the promo.
If there is any error during the process the promotion would be silently not applied.  Do we have any safeguard to just prevent the subscription being created if they have an introductary period but it came back as None?

Reply:
No we don't, this is a good point, we should put some thought into what we do in the general case here although it _should_ of course never happen!
I think this should be a different PR though

Reply:
yeah separate PR should be fine.   My question is really "how will we know if it's broken?".  Not from the type checker and not from the error logs, maybe  from the tests, and eventually from the users.